File: doctolib-frontend/src/services/appointment.service.ts
import { api } from "./api"
import type { CreateAppointmentDTO, UpdateAppointmentDTO, Appointment } from "../types/appointment"

export const appointmentService = {
  create: async (data: CreateAppointmentDTO): Promise<Appointment> => {
    const response = await api.post<Appointment>("/appointments/create", data)
    return response.data
  },

  update: async (id: string, data: UpdateAppointmentDTO): Promise<Appointment> => {
    const response = await api.put<Appointment>(`/appointments/${id}`, data)
    return response.data
  },

  getById: async (id: string): Promise<Appointment> => {
    const response = await api.get<Appointment>(`/appointments/${id}`)
    return response.data
  },

  getMine: async () => {
    const response = await api.get("/appointments/me")
    return response.data
  }
}

File: doctolib_mobile/lib/services/appointment_service.dart
import 'package:dio/dio.dart';
import '../api/dio_client.dart';

class AppointmentService {
  Future<Map<String, dynamic>> create(Map<String, dynamic> data) async {
    final dio = await DioClient.withToken();
    final response = await dio.post("/appointments/create", data: data);
    return response.data;
  }

  Future<Map<String, dynamic>> update(
    String id,
    Map<String, dynamic> data,
  ) async {
    final dio = await DioClient.withToken();
    final response = await dio.put("/appointments/$id", data: data);
    return response.data;
  }

  Future<Map<String, dynamic>> getById(String id) async {
    final dio = await DioClient.withToken();
    final response = await dio.get("/appointments/$id");
    return response.data;
  }

  Future<List<dynamic>> getMine() async {
    final dio = await DioClient.withToken();
    final response = await dio.get("/appointments/me");
    return response.data;
  }
}
